polygon/p2p/fetcher_base.go

```diff
@@ -63,12 +63,20 @@ type fetcher struct {
 	requestIdGenerator RequestIdGenerator
 }
 
-func (f *fetcher) FetchHeaders(ctx context.Context, start uint64, end uint64, peerId *PeerId) ([]*types.Header, error) {
+type FetcherResponse[T any] struct {
+	Data      T
+	TotalSize int
+}
+
+func (f *fetcher) FetchHeaders(ctx context.Context, start uint64, end uint64, peerId *PeerId) (FetcherResponse[[]*types.Header], error) {
 	if start >= end {
-		return nil, &ErrInvalidFetchHeadersRange{
-			start: start,
-			end:   end,
-		}
+		return FetcherResponse[[]*types.Header]{
```

Comment:
fyi you can simply do
```
return FetcherResponse[[]*types.Header]{}, ... 
```

all struct fields will be initialised with their "zero" values